observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.observability.interceptors;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.UnmodifiableListIterator;
+import io.grpc.MethodDescriptor;
+import io.grpc.observability.ObservabilityConfig;
+import io.grpc.observability.ObservabilityConfig.LogFilter;
+import io.grpc.observabilitylog.v1.GrpcLogRecord.EventType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class ConfigFilterHelper {
+  private static final Logger logger = Logger.getLogger(ConfigFilterHelper.class.getName());
+
+  private static final MethodFilterParams NO_FILTER_PARAMS
+      = new MethodFilterParams(false, 0, 0);
+
+  private final ObservabilityConfig config;
+  boolean methodOrServiceFilterPresent;
```

Comment:
Why not private? Also can be made final if initialized in constructor?

Reply:
Marked with `@VisibleForTesting`